src/onelogin/saml2/errors.py

```diff
@@ -50,3 +52,73 @@ def __init__(self, message, code=0, errors=None):
 
         Exception.__init__(self, message)
         self.code = code
+
+
+class OneLogin_Saml2_ValidationError(Exception):
+    """
+    This class implements another custom Exception handler, related
+    to exceptions that happens during validation process.
+    Defines custom error codes .
+    """
+
+    # Validation Errors
+    UNSUPPORTED_SAML_VERSION = 0
```

Comment:
I also don't like exception codes. I would like to have `OneLogin_Saml2_Error` that would be the base class for all exception raised by this library. Then we could have `OneLogin_Saml2_ValidationError` exception class inheriting that, and then each error would have own exception class.

Some code showing usage:

```python
class OneLogin_Saml2_Error(Exception):
    ....

class OneLogin_Saml2_ValidationError(OneLogin_Saml2_Error):
    # For backwards compatibility
    code = 123

class OneLogin_Saml2_ExpiredResponseError(OneLogin_Saml2_ValidationError):
    # If we want to keep backwards compatibility
    code = 123


# Usage:

try:
    auth.process_sso(...)
except OneLogin_Saml2_ExpiredResponseError as e:
    # Not important, tell user to retry
except OneLogin_Saml2_Error as e:
    # Inform user about the error and log it somewhere

```

With that said, I really liked the changes in this PR. 👍  I think that the exception handling can be changed to the way I showed later on without breaking any existing code, so that can be implemented later on.